repository_service_tuf_worker/repository.py

```diff
@@ -308,29 +315,42 @@ def _bump_and_persist(
         if persist:
             self._persist(role, role_name)
 
-    def _update_timestamp(self, snapshot_version: int) -> Metadata[Timestamp]:
+    def _update_timestamp(
+        self,
+        snapshot_version: Optional[int] = None,
+        skip: Optional[bool] = False,
+    ) -> Metadata[Timestamp]:
         """
         Loads 'timestamp', updates meta info about passed 'snapshot'
         metadata, bumps version and expiration, signs and persists.
 
         Args:
-            snapshot_version: snapshot version to add to new timestamp.
-            db_targets: RSTUFTarget DB objects will be changed as published in
-                the DB SQL.
+            snapshot_version: Optional snapshot version to new timestamp.
+            skip: Skip bumping version and expiration if snpashot
+            version is not provided
         """
         timestamp: Metadata[Timestamp] = self._storage_backend.get(
             Timestamp.type, None
         )
+        if not snapshot_version and skip:
+            logging.debug(
+                "No snapshot version and skip is True. Not bumping Timestamp"
+            )
+            return timestamp
+
         timestamp.signed.snapshot_meta = MetaFile(version=snapshot_version)
 
         self._bump_and_persist(timestamp, Timestamp.type)
 
+        logging.debug("Bumped version of 'Target' role")
         return timestamp
 
     def _update_snapshot(
```

Comment:
Honestly, this function isss really hard to read...
If you don't have the capacity or time to fix It, then don't but for sure we need a separate issue for this function specifically!
150 lines of different edge cases is hard to follow.

We had two cases that were hard to follow - `target_roles` and `bump_all` and now we add two more...
Given that we have 4 arguments and each of them has its own separate `if` meaning they are not exclusive branches then questions like: what will happen if I call 
_`_update_snapshot(target_roles=["project-a", "project-b"], only_target=True, only_snapshot=True]`_
happen?

Reply:
We have discussed refactoring this function for a while since you implemented the post-online rules bump ("bump_all").
While I was working on this PR, I started by refactoring this function, which resulted in a lot of changes in other parts that we don't want to make right now.
I think we can refactor it outside this PR. We can file an issue for it.

Reply:
Okay, as we are limited on time, lets do it in another pr after the conference.